src/librustc_parse/parser/diagnostics.rs

```diff
@@ -934,6 +934,25 @@ impl<'a> Parser<'a> {
             return self.expect(&token::Semi).map(drop);
         } else if !sm.is_multiline(self.prev_token.span.until(self.token.span)) {
             // The current token is in the same line as the prior token, not recoverable.
+        } else if sm.is_multiline(self.prev_token.span.until(self.token.span))
+            && [token::Comma, token::Colon].contains(&self.token.kind)
+            && &self.prev_token.kind == &token::CloseDelim(token::Paren)
```

Comment:
This seems a bit too constraining to a specific case. Let's see if the proposed change to the PR actually helps out for the case we are trying to address (without this code).

Reply:
Waiting to hear additional thoughts about this. This is still in place as of the comments in https://github.com/rust-lang/rust/pull/72348#discussion_r427719055